keras_cv/bounding_box/utils.py

```diff
@@ -107,12 +99,20 @@ def clip_to_image(bounding_boxes, bounding_box_format, images=None, image_shape=
     clipped_bounding_boxes = tf.where(
         tf.expand_dims(areas > 0.0, axis=-1), clipped_bounding_boxes, -1.0
     )
+
+    classes = tf.where(areas > 0.0, classes, tf.constant(-1, classes.dtype))
     nan_indices = tf.math.reduce_any(tf.math.is_nan(clipped_bounding_boxes), axis=-1)
-    clipped_bounding_boxes = tf.where(
-        tf.expand_dims(nan_indices, axis=-1), -1.0, clipped_bounding_boxes
+    classes = tf.where(nan_indices, tf.constant(-1, classes.dtype), classes)
+
+    # TODO update dict and return
+    clipped_bounding_boxes, classes = _format_outputs(
+        clipped_bounding_boxes, classes, squeeze
     )
-    clipped_bounding_boxes = _format_outputs(clipped_bounding_boxes, squeeze)
-    return clipped_bounding_boxes
+
+    result = bounding_boxes.copy()
```

Comment:
Why do we need to copy this?

Reply:
tf.function considerations.  KPLS call tf.cond() and if you dont copy the dictionary autograph complains that you are relying on inter-dependent branches.  copy() prevents this; 

BTW to be super clear, it only copies the dictionary not the tensors.